scripts/core/helpers/waitForMediaToBeReady.ts

```diff
@@ -1,9 +1,17 @@
 /**
  * @param elements An array of img, audio and video elements
- * @param eventName EventListener's name
  */
-export const waitForMediaToLoad = (elements: Array<any>, eventName: string): Promise<void> => new Promise((resolve) => {
-    let itemsLeftToLoad: number = elements.length;
+export const waitForMediaToLoad = (elements: Array<Element>): Promise<void> => new Promise((resolve) => {
```

Comment:
TypeScript checks aren't passing since `Element` doesn't have either a properly `complete` nor `readyState`.

I checked it again, and I think the following type signature is more accurate: `elements: Array<HTMLImageElement> | Array<HTMLVideoElement> | Array<HTMLAudioElement>)`

Create helpers like the one below to execute code specific to an element of a particular type.

```
export function isImage(x: Element): x is HTMLImageElement {
    return x.tagName === 'IMG';
}

if (isImage(document.querySelector('...'))) {
    // execute image specific code
}
```